app/view/Map.js

```diff
@@ -1,119 +1,138 @@
-                    zoom: 13,
-					styles: mystyles
-                }
-            }
-        ]
-    },
-    initialize: function(){
-        // Need this code, for more info on the process that works for our map so far, go to:
-        // http://www.joshmorony.com/integrating-the-google-maps-api-into-a-sencha-touch-application/
-        var me = this;
-        me.callParent(arguments);
-        this.initMap();
+		/* global setTimeout */
+		setTimeout(function () {
+			gMap.fitBounds(bounds);
+			gMap.panToBounds(bounds);
+			if(gMap.getZoom() > 15) { gMap.setZoom(15); }
+			if(gMap.getZoom() < 6) { gMap.setZoom(6); }
+		}, 100);
+	},
 
-        // TRYING TO MAKE MAP LOAD IN HERE
-        // Wasn't very successful
-        // var mapOptions = {
-        //     center: new google.maps.LatLng(44.634115, -124.062796),
-        //     mapTypeId: google.maps.MapTypeId.ROADMAP,
-        //     zoom: 18,
-        //     offsetWidth: 1
-        // };
-        // var map = new google.maps.Map(document.getElementById('map-canvas'), mapOptions);
+	center: function () {
+		var self = this;
+		var bounds = new google.maps.LatLngBounds();
+		for (var k = 0; k < self.markers.length; k++) {
+			bounds.extend(self.markers[k].position);
+		}
+		self.centerToBounds(bounds);
+	},
 
-        // **all of this code was never used**
-        // The fix that Sencha says is implemented (FYI: its not!)
-        // Also this fix doesn't work, because the map is not reinitialized when we go from home to list screen.
-        // var map = Ext.getCmp('SeaGrantMap').map;
-        // this.on({
-        //     show: function(){
-        //         google.maps.event.trigger(map, 'resize');
-        //         map.panTo(new google.maps.LatLng(44.634115, -124.062796));
-        //         google.maps.event.trigger(map, 'zoom_changed');
-        //     }
-        // }); 
-        // HELPER CODE 
-        // SEE: http://www.sencha.com/forum/showthread.php?151775-Ext.Map-rendered-incorrect-in-initially-hidden-Ext.tab.Panel
-        // google.maps.event.trigger(this.googleMap, 'resize');
-        // google.maps.event.trigger(this.googleMap, 'zoom_changed');
-        // this.googleMap.setCenter(coords);        
-    }, 
-    initMap: function(){
- 
-        var mapPanel = this.down('map');
-        WorkingWaterfronts.gMap = mapPanel.getMap();
+	currentInfoWindow: null,
+	markers: [],
 
-        // console.log('in the initmap function:');
-        // console.log(WorkingWaterfronts);
+	addPoint: function (point, id, text, onClick, onClose) {
+		var self	= this;
+		var gMap	= self.map();
+		onClick		= onClick || function () {};
 
-        // this should reset our map center, but it doesn't
-        //  setTimeout(function() {
-        //    WorkingWaterfronts.gMap.panTo(new google.maps.LatLng(44.634115, -124.062796));
-        // }, 100);
+		var html	= '<button onclick="Ext.Viewport.fireEvent(\'mapButton\',\''+id+'\')">' + text + '</button>';
```

Comment:
Please make sure that you comment this line: We are wrestling with scope here - how should we read this line?